pkgs/development/libraries/amdvlk/default.nix

```
@@ -62,14 +65,23 @@ stdenv.mkDerivation rec {
 
   cmakeDir = "../drivers/xgl";
 
+  # LTO is disabled in gcc for i686 as of #66528
+  cmakeFlags = if stdenv.system == "x86_64-linux" then [] else ["-DXGL_ENABLE_LTO=OFF"];
+
+  postPatch = if stdenv.system == "x86_64-linux" then "" else ''
+    sed -i 's#pal_setup_gcc_ipo()##' drivers/pal/cmake/PalCompilerOptions.cmake
+  '';
```

```suggestion
  postPatch = stdenv.lib.optionalString stdenv.is32bit ''
    substituteInPlace drivers/pal/cmake/PalCompilerOptions.cmake \
      --replace "pal_setup_gcc_ipo()" ""
  '';
```

`substituteInPlace` is preferred, unless regexps are really needed. Also warns when no substitutions are made.